### PR TITLE
Tech CORE-310 | Automatic Helm chart tags update

### PR DESCRIPTION
**Pull request by franciscobonand**
# Description

Kind: DevOps

Section: .gitlab/workflows and .gitlab/scripts

Summary: Created a GitHub action to automatically create a new release if a new tag is given. This action also build and updates Inspr CLI latest version, and updates the Helm Chart version as well.

## Changelog

> Created a new workflow / GitHub action
> Adapted old scripts that used to work on GitLab to work now in the GitHub action

## Pay attention to

> This is the first time I configure GitHub actions, not sure if it's really working

### CODE REVIEW

**Review by luiz-couto — APPROVED** [review]

File: .github/scripts/pushcli.sh [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
 
-echo $GCR_ACCOUNT_KEY
-
 gcloud auth configure-docker
-echo $GCR_ACCOUNT_KEY > $CI_PIPELINE_ID.json
-gcloud auth activate-service-account --key-file $CI_PIPELINE_ID.json
-export GOOGLE_APPLICATION_CREDENTIALS=$CI_PIPELINE_ID.json
+echo GCR_DETAILS > CI_RUN_ID.json
+gcloud auth activate-service-account --key-file CI_RUN_ID.json
+export GOOGLE_APPLICATION_CREDENTIALS=CI_RUN_ID.json
```
Comment (nicholaspcr) [inline .github/scripts/pushcli.sh]: do they work without the `$` symbol to get their value?

for example on line 7 wouldn't you be just echoing the `GCR_DETAILS` word into the `CI_RUN_ID.json` file?

File: .github/scripts/releasecli.sh [hunk 2]
```diff
@@ -1,9 +1,10 @@
+GITHUB_SHA=$1
 VERSION=$(cat version)
 BASE_URL="https://console.cloud.google.com/storage/browser/inspr-cli/bin"
 
 release-cli create --name "Release $VERSION" \
         --tag-name $VERSION \
-        --ref "$CI_COMMIT_SHA" \
+        --ref GITHUB_SHA \
```
Comment (nicholaspcr) [inline .github/scripts/releasecli.sh]: Same as before, does this actually work without the `$` before the `GITHUB_SHA`.

i tested doing something similar in my local machine and the value stored in the variable is only used when adding the `$`.

File: .github/scripts/releasecli.sh [hunk 3]
```diff
@@ -1,9 +1,10 @@
+GITHUB_SHA=$1
```
Comment (nicholaspcr) [inline .github/scripts/releasecli.sh:2]: Just realised that the `releasecli.sh` is the only shellscript that doesn't specify that is a bash script.

That is probably a mistake done by my part in a previous story but is interesting to fix this.

File: .github/scripts/pushcli.sh (hunk 1 above)
Comment (franciscobonand) [reply .github/scripts/pushcli.sh]: You're right

**Review by nicholaspcr — APPROVED** [review]

File: .github/scripts/releasecli.sh [hunk 4]
```diff
@@ -1,9 +1,11 @@
...
 VERSION=$(cat version)
 BASE_URL="https://console.cloud.google.com/storage/browser/inspr-cli/bin"
 
 release-cli create --name "Release $VERSION" \
         --tag-name $VERSION \
-        --ref "$CI_COMMIT_SHA" \
+        --ref $GITHUB_SHA \
         --assets-link "{\"name\":\"insprcli-linux-arm64\",
                         \"url\":\"$BASE_URL/insprcli-linux-arm64-$VERSION\"}" \
         --assets-link "{\"name\":\"insprcli-linux-amd64\",
```
Comment (ptcar2009) [inline .github/scripts/releasecli.sh:11]: Is does this CLI actually have the same API as gitlab's? I mean, if so, great, but did you check that?

File: .github/workflows/release.yml [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+        uses: google-github-actions/setup-gcloud@master
+
+      - name: Build CLI
+        run: bash .github/scripts/buildcli.sh
+
+      - name: Push CLI
+        run: bash .github/scripts/pushcli.sh ${{ secrets.GCR_ACCOUNT_KEY }} "${{ github.run_id }}"
+
+      - name: Release CLI
+        run: bash .github/scripts/releasecli.sh "${{ github.sha }}"
```
Comment (ptcar2009) [inline .github/workflows/release.yml]: This is what  may need a change in imaging.

File: .github/scripts/releasecli.sh (hunk 4 above)
Comment (franciscobonand) [reply .github/scripts/releasecli.sh:11]: As we discussed, we'll be storing the CLI binaries in the GitHub Release

**Review by LucasStarlingdePaulaSalles — APPROVED** [review]

**Pull request merged**